File: src/main/java/com/example/backendthuvien/entity/Order.java
package com.example.backendthuvien.entity;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "fullname",length = 100)
    private String fullName;

    @ManyToOne @JoinColumn(name = "user_id",referencedColumnName = "id")
    private User user;

    @Column(name = "email",length = 200)
    private String email;

    @Column(name = "phone_number",length = 12,nullable = false)
    private String phoneNumber;


    @Column(name = "address",length = 200)
    private String address;



    @Column(name = "note",length = 100)
    private String note;

    @Column(name = "order_date")
    private Date orderDate;

    @Column(name = "status")
    private String status;

    @Column(name = "total_money")
    private float 	totalMoney;

    @Column(name = "shipping_method")
    private String shippingMethod;

    @Column(name = "shipping_address")
    private String shippingAddress;

    @Column(name = "shipping_date")
    private LocalDate shippingDate;

    @Column(name = "tracking_number")
    private String trackingNumber;

    @Column(name = "payment_method")
    private String paymentMethod;

    @Column(name = "active")
    private Boolean active;//thuộc về admin

@OneToMany (mappedBy = "order",cascade = CascadeType.ALL,fetch = FetchType.LAZY)
    @JsonManagedReference
    private List<Order_Detail> orderDetails;

    @ManyToOne @JoinColumn(name = "coupon_code", referencedColumnName = "coupon_code")
    private Coupon coupon;



}

File: src/app/responses/user/order.ts
export interface Order {
  id: number;
    user_id: number;           // ID người dùng
  fullname: string;          // Họ tên
  email: string;             // Địa chỉ email
  phone_number: string;      // Số điện thoại
  address: string;           // Địa chỉ giao hàng
  note: string;              // Ghi chú
  total_money: number;       // Tổng tiền
  shipping_method: string;   // Phương thức vận chuyển
  payment_method: string;    // Phương thức thanh toán
  coupon_code: string;       // Mã giảm giá (nếu có)
  active:string;
  order_date: Date;
  cart_items: {              // Danh sách sản phẩm trong giỏ hàng
    product_id: number;      // ID sản phẩm
    quantity: number;        // Số lượng
  }[];

}